IPython/core/tests/test_interactiveshell.py

```diff
@@ -546,6 +546,10 @@ def test_exit_code_signal_csh(self):
             else:
                 del os.environ['SHELL']
 
+    def tearDown(self):
```

Comment:
Isn't the right fix to re-order the inheritance so the Mixin comes first?

Reply:
> I can retry though.

Hum, some seem to work, It may have been a random failure when I tried. I'll re-check and do that where it works.

Reply:
Pfff, PEBKAC, I was testing without #11174 merged and so was still seing ResourceWarnings.